sparkcore/src/main/scala/quasar/physical/sparkcore/fs/local/package.scala

```
@@ -0,0 +1,95 @@
+import quasar.effect._
+import quasar.fs.ReadFile.ReadHandle
+import quasar.fs.WriteFile.WriteHandle
+import quasar.fp.TaskRef
+import quasar.fp.free._
+
+import java.io.PrintWriter
+
+import pathy.Path._
+import org.apache.spark._
+import scalaz._, Scalaz._
+import scalaz.concurrent.Task
+
+package object local {
+  val FsType = FileSystemType("sparklocal")
+
+  type Eff0[A] = Coproduct[KeyValueStore[ReadHandle, SparkCursor, ?], Read[SparkContext, ?], A]
+  type Eff1[A] = Coproduct[KeyValueStore[WriteHandle, PrintWriter, ?], Eff0, A]
+  type Eff2[A] = Coproduct[Task, Eff1, A]
+  type Eff[A] = Coproduct[MonotonicSeq, Eff2, A]
+
+  final case class SparkFSDef[S[_]](run: Free[Eff, ?] ~> Free[S, ?], close: Free[S, Unit])
+
+  def sparkFsDef[S[_]](sparkConf: SparkConf)(implicit
+    S0: Task :<: S
+  ): Free[S, SparkFSDef[S]] = {
+
+    val genSc = Task.delay {
+      new SparkContext(sparkConf.setAppName("quasar"))
+    }
+
+    lift((TaskRef(0L) |@|
+      TaskRef(Map.empty[ReadHandle, SparkCursor]) |@|
+      TaskRef(Map.empty[WriteHandle, PrintWriter]) |@|
+      genSc) {
+      (genState, sparkCursors, printWriters, sc) =>
+        val taskInter: Eff ~> Task = MonotonicSeq.fromTaskRef(genState) :+:
+          NaturalTransformation.refl[Task] :+:
+          KeyValueStore.fromTaskRef[WriteHandle, PrintWriter](printWriters) :+:
+          KeyValueStore.fromTaskRef[ReadHandle, SparkCursor](sparkCursors) :+:
+            Read.constant[Task, SparkContext](sc)
+
+        val sInter: Eff ~> S = taskInter andThen injectNT[Task, S]
+
+      SparkFSDef(mapSNT[Eff, S](sInter), lift(Task.delay(sc.stop())).into[S])
+    }).into[S]
+  }
+
+  def definition[S[_]](implicit S0: Task :<: S, S1: PhysErr :<: S):
+      FileSystemDef[Free[S, ?]] =
+    FileSystemDef.fromPF {
+      case (FsType, uri) =>
+        val fsConf = parseUri(uri).getOrElse(SparkFSConf(new SparkConf().setMaster("local[*]"), rootDir </> dir("rabbit")))
```

If `parseUri` fails it should be an error rather than supplying this default config, which will confuse users.